clang_bind/parse.py

```diff
@@ -38,40 +51,99 @@ def __init__(self, file, compiler_arguments):
         )
         self.filename = source_ast.spelling
         self.tree = Tree()
-        self.root_node = self.tree.create_node(
-            identifier=Node(source_ast.cursor), tag=repr(Node(source_ast.cursor))
-        )
+        parsed_info = ParsedInfo(source_ast.cursor)
+        self.root_node = self.tree.create_node(tag=repr(parsed_info))
+        self._parsed_info_map[self.root_node.identifier] = parsed_info
+        self.__construct_tree(self.root_node)
 
     @staticmethod
-    def is_node_from_file(node, filename):
-        """
-        Check if the node belongs in the file.
-        """
-        return node.location.file and node.location.file.name == filename
+    def is_cursor_in_file(cursor, filename):
+        """Checks if the cursor belongs in the file.
 
-    def _is_valid_child(self, child_cursor):
-        """
-        Check if the child is valid (child should be in the same file as the parent).
+        :param cursor: An object of :class:`clang.cindex.Cursor`
+        :type cursor: class:`clang.cindex.Cursor`
+        :param filename: Filename to search the cursor
+        :type filename: str
+        :return: `True` if cursor in file, else `False`
+        :rtype: bool
         """
-        return self.is_node_from_file(child_cursor, self.filename)
+        return cursor.location.file and cursor.location.file.name == filename
+
+    def __is_valid_child(self, child_cursor):
+        """Checks if the child is valid (child should be in the same file as the parent).
 
-    def _construct_tree(self, node):
+        :param child_cursor: The child cursor to check, an object of :class:`clang.cindex.Cursor`
+        :type child_cursor: class:`clang.cindex.Cursor`
+        :return: `True` if child cursor in file, else `False`
+        :rtype: bool
         """
-        Recursively generates tree by traversing the AST of the node.
+        return self.is_cursor_in_file(child_cursor, self.filename)
+
+    def __construct_tree(self, node):
```

Comment:
Why __construct_tree instead of _construct_tree?

Reply:
I thought double underscore was the standard. Changed.